sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpElementAccessTracker.cs

```diff
@@ -55,5 +55,28 @@ public CSharpElementAccessTracker(IAnalyzerConfiguration analyzerConfiguration,
                     constantValue.Value is string constant &&
                     constant == value;
             };
+
+        #region Syntax-level checking methods
+
+        public override ElementAccessCondition MatchSetter() =>
+            (context) => ((ExpressionSyntax)context.Expression).IsLeftSideOfAssignment();
+
+        public override ElementAccessCondition MatchProperty(MemberDescriptor member)
+        {
+            return (context) => {
+                var accessSyntax = context.Expression as ElementAccessExpressionSyntax;
+                if(accessSyntax.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+                {
+                    var memberAccessSyntax = accessSyntax.Expression as MemberAccessExpressionSyntax;
```

Comment:
and you can chain this with `is` inside the above if statement

Reply:
actually I tend to think all of this can be shortened into a single arrow-body function like
```
public override ElementAccessCondition MatchProperty(MemberDescriptor member) =>
  (context) =>
    context.Expression is ElementAccessExpressionSyntax accessSyntax &&
    accessSyntax.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression) &&
    accessSyntax.Expression is MemberAccessExpressionSyntax memberAccessSyntax &&
    memberAccessSyntax.Name.Identifier.ValueText is SyntaxNode propertyName &&
    context.SemanticModel.GetTypeInfo(memberAccessSyntax.Expression) is ITypeInfo enclosingClassType &&
    enclosingClassType.Type != null && 
    member.IsMatch(propertyName, enclosingClassType.Type, false);
```

Reply:
applies both to C# and VB impl

Reply:
I added many checks and switched to single-arrow function, so as to be on the safer side. Only the 1st one is not necessary, as the first cast is based on the fact that we know the type of node the checker is registered on